src/Device/Driver/ATR833.cpp

```diff
@@ -77,14 +123,170 @@ class ATRBuffer {
 };
 
 bool
-ATR833Device::DataReceived(std::span<const std::byte>,
+ATR833Device::DataReceived(std::span<const std::byte> s,
                            [[maybe_unused]] NMEAInfo &info) noexcept
 {
-  // actually made no use of radio information
-  // TODO: interpret data delivered by ATR833 to display Radio settings...
+  assert(!s.empty());
+  const auto *data = s.data();
+  const auto *const end = data + s.size();
+
+  do {
+    // Append new data to the buffer, as much as fits in there
+    auto range = rx_buf.Write();
+    if (rx_buf.IsFull()) {
+      // Overflow: reset buffer to recover quickly
```

Comment:
If the buffer is full at this point, it has already overflowed, and if clearing it is the way out, it should have been done already.